DemoQA/Tests/elements_webtables.js

```diff
@@ -20,72 +21,91 @@ test(`Check page validity`, async t => {
         .expect(currentUsersList).eql(latestUsersList);
 });
 
 
 test(`Edit a new user`, async t => {
     await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
-    await registrationFormPage.addNewUser();
+    await webTablesPage.addNewUser(myUserDetails);
 
     const newUser = new UserInfoRow(3);
 
     await newUser.editUser();
-    await registrationFormPage.clearFormFields(); 
-    await registrationFormPage.fillRegistrationForm(newUserDetails);
-    await registrationFormPage.submitForm();
+    await registrationFormPage.fillRegistrationForm(myUserDetails);
 
     const changedUserDetails = await newUser.getUserDetails();
 
     await t
-        .expect(changedUserDetails.firstName).eql(newUserDetails.firstName)
-        .expect(changedUserDetails.lastName).eql(newUserDetails.lastName)
-        .expect(changedUserDetails.age).eql(newUserDetails.age)
-        .expect(changedUserDetails.emailAddress).eql(newUserDetails.email)
-        .expect(changedUserDetails.salary).eql(newUserDetails.salary)
-        .expect(changedUserDetails.department).eql(newUserDetails.department);
+        .expect(changedUserDetails.firstName).eql(myUserDetails.firstName)
+        .expect(changedUserDetails.lastName).eql(myUserDetails.lastName)
+        .expect(changedUserDetails.age).eql(myUserDetails.age)
+        .expect(changedUserDetails.emailAddress).eql(myUserDetails.email)
+        .expect(changedUserDetails.salary).eql(myUserDetails.salary)
+        .expect(changedUserDetails.department).eql(myUserDetails.department);
 });
 
 
 test(`Edit an existing user`, async t => {
     await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
-    await webTablesPage.selectAnyUserRow();
 
+    const existingUser = new UserInfoRow(0);
+
+    await existingUser.editUser();
+    await registrationFormPage.fillRegistrationForm(myUserDetails);
 
+    const modifiedUserDetails = await existingUser.getUserDetails();
 
-//edit user details , specifci but general at the same time INSANITY 
-//expect new detail in new cell to be X
-// check previous tests - you know how to do this.
-//exoect oher details in other cells to be same 
+    await t
+    .expect(modifiedUserDetails.firstName).eql(myUserDetails.firstName)
+    .expect(modifiedUserDetails.lastName).eql(myUserDetails.lastName)
+    .expect(modifiedUserDetails.age).eql(myUserDetails.age)
+    .expect(modifiedUserDetails.emailAddress).eql(myUserDetails.email)
+    .expect(modifiedUserDetails.salary).eql(myUserDetails.salary)
+    .expect(modifiedUserDetails.department).eql(myUserDetails.department);
+});
 
-});
\ No newline at end of file
+
+test(`Search table for firstName`, async t => {
+    await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
+    
+    const firstNameUser = new UserInfoRow(0);
+    const userName = "Cierra"; // should it be more general? so it catches any first Name property from table?
```

Comment:
Yes, it should be. This approach will work only here and only now. On a real project you will need a more sustainable approach like:

1. Count the users on your list
2. Get a random user from from 0 to your maximum number
3. Get those user's details
4. Search for that user's fistname/lastname/age/email/etc
5. Expect the number of users after search to be less than the max. Ideally you get 1 user, but that's also prone to failure
6. Expect for every user found to have the search query in their details.